### PR TITLE
Add #[must_use] to len and is_empty

### PR DESCRIPTION
**Pull request by jkugelman**
Parent issue: #89692

r? @joshtriplett

### CODE REVIEW

**Comment by the8472** [comment]
Looks like this failed a rollup in https://github.com/rust-lang/rust/pull/89801#issuecomment-940674499

@bors r-

File: src/tools/clippy/tests/ui/iter_count.rs [hunk 1]
```diff
@@ -51,36 +51,36 @@ fn main() {
...
+    let _ = binary_heap.into_iter().count();
 
     // Make sure we don't lint for non-relevant types.
     let false_positive = HasIter;
-    false_positive.iter().count();
-    false_positive.iter_mut().count();
-    false_positive.into_iter().count();
+    let _ = false_positive.iter().count();
+    let _ = false_positive.iter_mut().count();
+    let _ = false_positive.into_iter().count();
```
[the remark this answers is not in the document]
Comment (joshtriplett) [reply src/tools/clippy/tests/ui/iter_count.rs:85]: I don't have a strong preference, so go ahead and switch this one to suppress the lint.

[the remark this answers is not in the document]
Comment (jkugelman) [reply src/tools/clippy/tests/ui/iter_count.rs]: Done.

**Pull request merged**